nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java

```diff
@@ -90,19 +89,20 @@ public Map<Integer, Nullness> onOverrideMethodInvocationParametersNullability(
         optimizedLibraryModels.explicitlyNullableParameters(methodSymbol);
     ImmutableSet<Integer> nonNullParamsFromModel =
         optimizedLibraryModels.nonNullParameters(methodSymbol);
-    Sets.SetView<Integer> intersection =
```

Comment:
This sanity check was costly, particularly the call to `String.format()`.  I changed the code to implicitly compute the intersection and to only call `String.format()` when a violation is detected

Reply:
Right, I guess one annoying issue with `Preconditions.checkX(...)` vs `if (X) { throw new ... }` is that it eagerly produces the error message for things that haven't happened...

p.s. re-ArrayList-vs-Map, one of the things I thought we could get away with by using `Map` was to have sparse data there, e.g. for methods with a ton of primitive arguments, but I think that, having seen now the full implementation, in most cases we have no choice but to either add the default values for all arguments before sending it through the handler interface or to perform some sort of double filter (i.e. check while generating the map what arguments we don't need explicit nullness data for, then at the end make sure to ignore nullness values that are in those positions in the map... which is probably more trouble than it's worth).

Reply:
Yeah, I see there could be some waste for a method with a lot of primitive-type arguments.  When I get time I will tinker with the representation more (maybe even a `Nullness[]` makes sense since we know the max size ahead of time).  We need to iterate over this data structure, so having something that supports sequential access might be a win in the common case.

Reply:
`Nullness[]` makes sense to me. It could probably even be a bit-vector, if we really wanted to squeeze every last ms of performance out of it (though I'd argue against it due to code maintainability). 

Let me review this is as is, land it if we can, then cut the next nullaway release. After that, I can make the `Nullness[]` change if you want, so long as you help me test it for performance.

Reply:
SGTM.  And first let me try to re-collect profiles to be more sure about where to optimize.